src/hooks/useCountries.js

```diff
@@ -0,0 +1,26 @@
+import { useEffect, useState } from 'react';
+import { getCountries } from '../services/country.js';
+
+export function useCountries() {
+  const [countries, setCountries] = useState([]);
+  const [type, setType] = useState(['Antartica']);
+  const [error, setError] = useState('');
+
+  useEffect(() => {
+    async function fetchData() {
+      try {
+        const data = await getCountries();
+        setCountries(data);
+      } catch (e) {
+        setError(e.text);
+      }
+    }
+    fetchData();
+  }, []);
+
+  const countryFilter = () => {
+    if (type === 'all') return countries;
+    return countries.filter((arr) => arr.continent === type);
```

Comment:
state in React should be immutable — the only way to change state should be using a setState function. here you're taking that state, mutating it, then directly using that to render rather than using setState to rerender